Failure to build docs from main because of pydantic's ModelMetaclass

## 🐛 Bug
`make docs` fails with napari/napari main and napari/docs main with the following error:

```
make docs
echo clean
clean
echo /home/melissa/projects/napari-docs/
/home/melissa/projects/napari-docs/
rm -rf /home/melissa/projects/napari-docs/docs/_build/
rm -rf /home/melissa/projects/napari-docs/docs/api/napari*.rst
rm -rf /home/melissa/projects/napari-docs/docs/gallery/*
rm -rf /home/melissa/projects/napari-docs/docs/_tags
python -m pip install -qr /home/melissa/projects/napari-docs/requirements.txt
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
numpydoc 1.6.0 requires sphinx>=5, but you have sphinx 4.5.0 which is incompatible.
python /home/melissa/projects/napari-docs/docs/_scripts/prep_docs.py
Cloning into '/home/melissa/projects/napari-docs/npe2'...
remote: Enumerating objects: 2491, done.
remote: Counting objects: 100% (783/783), done.
remote: Compressing objects: 100% (231/231), done.
remote: Total 2491 (delta 619), reused 600 (delta 540), pack-reused 1708
Receiving objects: 100% (2491/2491), 722.42 KiB | 5.27 MiB/s, done.
Resolving deltas: 100% (1661/1661), done.
Note: switching to 'tags/v0.7.3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 9d29e4d Add pydantic 2 compatibility layer (#321)
Rendered /home/melissa/projects/napari-docs/docs/plugins/_npe2_manifest.md
Rendered /home/melissa/projects/napari-docs/docs/plugins/_npe2_writers_guide.md
Rendered /home/melissa/projects/napari-docs/docs/plugins/_npe2_readers_guide.md
Rendered /home/melissa/projects/napari-docs/docs/plugins/_npe2_sample_data_guide.md
Rendered /home/melissa/projects/napari-docs/docs/plugins/_npe2_widgets_guide.md
Rendered /home/melissa/projects/napari-docs/docs/plugins/_npe2_contributions.md
Traceback (most recent call last):
  File "/home/melissa/projects/napari-docs/docs/_scripts/prep_docs.py", line 38, in <module>
    main()
  File "/home/melissa/projects/napari-docs/docs/_scripts/prep_docs.py", line 33, in main
    __import__('update_preference_docs').main()
  File "/home/melissa/projects/napari-docs/docs/_scripts/update_preference_docs.py", line 4, in <module>
    from pydantic.main import ModelMetaclass
ImportError: cannot import name 'ModelMetaclass' from 'pydantic.main' (/home/melissa/micromamba/envs/napari-dev/lib/python3.10/site-packages/pydantic/main.py)
make: *** [Makefile:27: prep-docs] Error 1
```

Using pip list, I get 
```
pydantic                      2.4.2
pydantic_core                 2.10.1
```

The failure indicates there is an issue with the `update_preference_docs.py` script where we see an attempt to import `ModelMetaclass`. I see that this has been made private by pydantic: https://github.com/pydantic/pydantic/issues/6381#issuecomment-1619296261

Any thoughts on how to proceed?

--- a/docs/_scripts/update_preference_docs.py
+++ b/docs/_scripts/update_preference_docs.py
@@ -1,7 +1,7 @@
 from pathlib import Path
 
 from jinja2 import Template
-from pydantic.main import ModelMetaclass
+from napari._pydantic_compat import ModelMetaclass
 from qtpy.QtCore import QTimer
 from qtpy.QtWidgets import QMessageBox
 

